app/waf/src/models/wafpolicy.model.ts

```diff
@@ -45,8 +47,32 @@ export class Wafpolicy extends CommonEntity {
   })
   public: boolean;
 
+  @property({
+    type: 'string',
+    required: false,
+    schema: {
+      response: true,
+    },
+  })
+  state?: string;
+
   constructor(data?: Partial<Wafpolicy>) {
     super(data);
     this.as3Class = 'WAF_Policy';
   }
 }
+
+export class WafpolicyOnDevice extends Wafpolicy {
+  @property({
+    type: 'string',
+    required: false,
+    schema: {
+      response: true,
+    },
+  })
+  state?: string;
```

Comment:
Maybe we can remove "?" and add a default value. It depends on you.

Reply:
I think it's better not. It may be confused in some status and leads to debug difficulties.